### ISSUE TITLE
Consider: tryCDN System Setting

### ISSUE THREAD

**Issue by jpdevries** [modxcms/revolution#13553]
### Summary
There are advantages to loading common assets such as jQuery or React from a CDN. Mainly, doing so leverages the browser cache across MODX installations. Currently if you log into 20 MODX sites you download the same dependencies 20 times. If Extras loaded common assets from a CDN you'd only download them once. This not only makes the user experience snappier, but it makes MODX more accessible to economically disparaged users who need to be mindful of their bandwidth usage.

As discussed in https://github.com/modxcms/revolution/issues/12647 CDNs aren't totally reliable so they shouldn't be entirely relied on. Local fallbacks should be used. I've written about this here:
https://github.com/jpdevries/modx-style-guide#lazy-load-scripts

I'm all for CDNs but it is important to allow them to be opted out of so that users whom expect CDNs to be unreachable don't have to wait for requests to time out before local fallbacks are triggered. Also, service workers can only offline&ndash;cache assets from the same domain so you may set `[[++tryCDN]]` to false and prime your cache if you know you are going to be offline for a bit.

### Step to reproduce
n/a

### Observed behavior
The Management experience could be snappier if the browser cache were able to be leveraged across MODX installations. 

### Expected behavior
If `[[++tryCDN]]` is true Extra Developers could tap into this setting and decide whether or not they should try to load assets from a CDN. 

### Environment
Recommending to consider this for MODX 2.6 or MODX 3.0

### See Also
https://github.com/modxcms/revolution/issues/12647

### Example:
```js
import { join } from 'path';
import lazyLoadScript from 'lazyload-script';

  if(!window.React) promises.push(
    // if needed, try to load React from a CDN (if allowed), fallback to a local copy
    (MODx.tryCDN) ? lazyLoadScript(`https://cdnjs.cloudflare.com/ajax/libs/react/${reactVer}/react.min.js`, `react.${reactVer}.min.js`).catch((err => (
                     lazyLoadScript(join(MODx.assets_url, `./js/vendor/react.${reactVer}.min.js`), `react.${reactVer}.min.js`)
    ))) : lazyLoadScript(join(MODx.assets_url, `./js/vendor/react.${reactVer}.min.js`), `react.${reactVer}.min.js`)
  );
```

**Comment by OptimusCrime** [comment]
Is the goal to load just libraries from CDN, or the MODX code too?

If this is included into 2.6 it would need to be opt-in, not opt-out.

**Comment by JoshuaLuckers** [comment]
At the moment we are not using a CDN therefore I'm closing this issue.
